lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -183,13 +183,18 @@ defmodule Mix.Tasks.Escript.Build do
       end)
       |> parse_strip_beams_options()
 
-    escript_mod = String.to_atom(Atom.to_string(app) <> "_escript")
+    escript_mod = String.to_atom(Atom.to_string(app || project[:app]) <> "_escript")
```

Comment:
Note: the configuration may set `app: nil`, in which case it ended up generating `nil_escript.beam`.

Reply:
I'd say that's fine, we shouldn't work around that.

```suggestion
    escript_mod = String.to_atom(Atom.to_string(app) <> "_escript")
```